src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1193,8 +1313,6 @@ private static ImmutableArray<StateForCase> RemoveEvaluation(ImmutableArray<Stat
                                 }
                             }
                             break;
```

Comment:
Do these removals actually affect anything? If not, consider keeping them.

Reply:
>Do these removals actually affect anything? If not, consider keeping them.

No, that's exactly why they're removed. Would you prefer to do this in a separate PR instead?

Reply:
> Would you prefer to do this and others in a separate PR instead?

Yes. This is a big PR. Let's keep any refactorings separate.